aiosmtpd/controller.py

```diff
@@ -425,7 +425,10 @@ def _trigger_server(self):
         with ExitStack() as stk:
             s = stk.enter_context(create_connection((hostname, self.port), 1.0))
             if self.ssl_context:
-                s = stk.enter_context(self.ssl_context.wrap_socket(s))
+                context = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+                context.check_hostname = False
+                context.verify_mode = ssl.CERT_NONE
+                s = stk.enter_context(context.wrap_socket(s))
```

Comment:
Finally able to take a look at this, and... it's definitely a bad idea to just hard code False in for checking hostname.